lib/apb.ts

```diff
@@ -210,19 +212,45 @@ export class apb {
     }
   }
 
+  generateParametersForSoclessTask(state_name: string, handle_state_kwargs: Record<string, any>){
+    const parameters : SoclessTaskStepParameters = {
+      "execution_id.$" : "$.execution_id",
+      "artifacts.$" : "$.artifacts",
+      "errors.$" : "$.errors",
+      "results.$" : "$.results",
+      "State_Config" : {
+        "Name" : state_name,
+        "Parameters" : handle_state_kwargs
+      },
+    }
+    return parameters
+  }
+
+  //! Interactions still need helper states ):
+  generateParametersForSoclessInteraction(state_name: string, handle_state_kwargs: Record<string, any>, function_name: string){
+    const parameters = {
+      FunctionName: function_name,
+      Payload: {
+        "sfn_context" : this.generateParametersForSoclessTask(state_name, handle_state_kwargs),
+        "task_token.$": "$$.Task.Token",
+      },
+    }
+    return parameters
+  }
+
   transformTaskState(stateName: string, stateConfig, States, DecoratorFlags) {
     let output = {}
     let newConfig = Object.assign({}, stateConfig)
-    if (!!stateConfig['Next']) {
-      Object.assign(newConfig, { Next: this.resolveStateName(stateConfig.Next, States) })
+    if (!!newConfig['Next']) {
+      Object.assign(newConfig, { Next: this.resolveStateName(newConfig.Next, States) })
     }
 
-    if (!!stateConfig['Catch']) {
-      Object.assign(newConfig, { Catch: this.transformCatchConfig(stateConfig.Catch, States) })
+    if (!!newConfig['Catch']) {
+      Object.assign(newConfig, { Catch: this.transformCatchConfig(newConfig.Catch, States) })
     }
 
-    if (!!stateConfig['Retry']) {
-      Object.assign(newConfig, { Retry: this.transformRetryConfig(stateConfig.Retry, stateName) })
+    if (!!newConfig['Retry']) {
+      Object.assign(newConfig, { Retry: this.transformRetryConfig(newConfig.Retry, stateName) })
```

Comment:
This looks correct but we should tread carefully here. Since the variable we're referencing in our if checks is the same variable we're mutating, a future change to this code can easily introduce a hard to discover bug